.github/workflows/node_modules-pull_request.yml

```diff
@@ -0,0 +1,67 @@
+name: node_modules
+on: pull_request
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    timeout-minutes: 2
+    outputs:
+      package_json: ${{ steps.package_json.outputs.result }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Read package.json
+        id: package_json
+        uses: actions/github-script@v3
+        with:
+          script: return require('fs').readFileSync('package.json');
+          result-encoding: string
+
+      - name: Resolve and verify node_modules gitlink
+        id: verify
+        run: |
+            set -ux
+            node_modules="$(git rev-parse '${{ github.sha }}:node_modules')"
+            test -n "${node_modules}"
+            git fetch "${GITHUB_SERVER_URL}/${GITHUB_REPOSITORY%/*}/node-cache" "${node_modules}"
+            git cat-file blob "${node_modules}:.package.json" > node_modules-package.json
+            diff -u node_modules-package.json package.json
```

Comment:
Could this re-use tools/node-modules with some test-static-code or similar command?

Reply:
I'm trying to keep the workflow completely independent from any file other than `package.json`.